staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -244,29 +244,36 @@ func (s *SpdyRoundTripper) dialWithoutProxy(ctx context.Context, url *url.URL) (
 		}
 	}
 
+	// dialWithoutProxy either connects directly to the host specified by url or to a proxy.
+	// In the latter case we don't want cluster-specific TLS options such as tls-server-name to be applied.
+	tlsConfig := s.tlsConfig
+	if s.proxier != nil {
+		tlsConfig = &tls.Config{}
```

Comment:
I can buy we don't want `tls-server-name` applied to the connection to the proxy... I'm less sure about the CA bundle and the InsecureSkipVerify setting

Reply:
also, doesn't this lose MinVersion/NextProtos options we populate?

Reply:
Thanks for the review!

As far as I know the following is true in most cases: the CA bundle configured in the cluster configuration only contains the cluster CA. HTTPS_PROXY is a system-wide configuration and uses a publicly or company-wide trusted CA. Our solution here right now is that we concatenate the system CA bundle and the cluster CA - but this is high maintenance: we need to update the cluster configuration each time the system CA bundle changes.

curl offers separate flags for InsecureSkipVerify: `--insecure` and `--proxy-insecure`.

From my point of view HTTPS_PROXY is a system-wide thing and it is unexpected/unwanted if any cluster-related option affects connections to it. 

However, there might be a solution that would be less disruptive and still fix our issues: we could clone the TLS config and just unset the `ServerName` field. In addition, there is already an option that allows to specify the path to a CA bundle, but it doesn't work simultaneously with the embedded CA bundle option. If we could make both work at the same time we could point one to the system CA bundle and store the cluster CA in the other.

https://github.com/kubernetes/kubernetes/blob/37da905c0c673c9cb07ca724384d37e725602a0c/staging/src/k8s.io/client-go/tools/clientcmd/validation.go#L237

Reply:
Hi @liggitt - can I do anything to help this move forward? How would a satisfying solution look like?

Reply:
since this is a low-level library, we have to ensure that the following scenarios are not broken by this change:
* we don't want to start failing TLS verification for https proxy users for whom it is currently working
* we don't want to start skipping TLS verification for https proxy users incorrectly

the types of things we could do to make progress here:
* make the changes here require opt-in, so all existing consumers are unaffected
* prove that any changes made here that are not opt-in *only* change behavior in cases that previously errored incorrectly